cpp/src/neighbors/brute_force.cu

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuvs/neighbors/brute_force-inl.hpp>
```

Comment:
Let's consolidate `inl` and `ext` header files now that we're moving from header-only to strictly compiled design. Please remember that these heade rfiles should not have any instantiations in them, and so all that should be getting defined is the class interface. We should be instantiating and implementing the interfaces in dedicated cpp files.